cmd/subctl/execute/execute.go

```diff
@@ -75,3 +75,15 @@ func IfSubmarinerInstalled(run OnClusterFn) OnClusterFn {
 		return run(clusterInfo, status)
 	}
 }
+
+func IfServiceDiscoveryInstalled(run OnClusterFn) OnClusterFn {
+	return func(clusterInfo *cluster.Info, status reporter.Interface) bool {
+		if clusterInfo.ServiceDiscovery == nil {
+			status.Warning(constants.ServiceDiscoveryNotInstalled)
```

Comment:
Do we always want to warn that service discovery is not installed? In some scenarios this would be perfectly fine, but some users may consider the warning to be an actual warning and indicative of a problem with their setup.

Reply:
This is intended for top-level commands so I think warning makes sense over success or failure. This is what we do Submariner not installed. We don't have an info message type.